### DISCUSSION TITLE
Conflicting Peer Dependencies

### DISCUSSION THREAD

**Discussion by woltob** [mikro-orm/nestjs#43] [Q&A]
Hi,

I am currently upgrading NestJS to version 8 as it is supposed to work now with MikroORM.

However, I am receiving the following issues after a clean npm install and running npm audit fix.

```
npm audit fix
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @mikro-orm/nestjs@4.3.0
npm ERR! Found: @nestjs/common@8.0.11
npm ERR! node_modules/@nestjs/common
npm ERR!   @nestjs/common@"^8.0.11" from the root project
npm ERR!   peer @nestjs/common@"^7.0.0||^8.0.0" from @mikro-orm/nestjs@4.3.0
npm ERR!   node_modules/@mikro-orm/nestjs
npm ERR!     @mikro-orm/nestjs@"^4.3.0" from the root project
npm ERR!   10 more (@nestjs/config, @nestjs/core, @nestjs/event-emitter, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @nestjs/common@"^7.0.0||^8.0.0" from @mikro-orm/nestjs@4.3.0
npm ERR! node_modules/@mikro-orm/nestjs
npm ERR!   @mikro-orm/nestjs@"^4.3.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @nestjs/common@7.5.5
npm ERR! node_modules/@nestjs/common
npm ERR!   peer @nestjs/common@"^7.0.0||^8.0.0" from @mikro-orm/nestjs@4.3.0
npm ERR!   node_modules/@mikro-orm/nestjs
npm ERR!     @mikro-orm/nestjs@"^4.3.0" from the root project
```

As far as I would understand it, peer @nestjs/common@"^7.0.0||^8.0.0" from @mikro-orm/nestjs@4.3.0 would also include version 8.0.11 which I have installed (@nestjs/common)

Any idea on how to fix this so that it does not show up anymore?

Was running `npm audit fix --legacy-peer-deps`to resolve it for  now.

**Comment by B4nan** [comment]
This feels like you still have `@nestjs/common@7.5.5` installed in your project, I don't think it has anything to do with the ORM setup. Maybe try to wipe your node modules, remove lock file and reinstall.

This module is not dependent on nest directly (only as a peer dep), so this `@nestjs/common@7.5.5` needs to come from your own project.
